tests/suites/os/tests/modem/index.js

```diff
@@ -198,15 +198,15 @@ module.exports = {
 
                     test.comment(`signal-quality: ${modemData.modem.generic['signal-quality'].value}, state: ${modemData.modem.generic.state}`)
 
-                    // ping test endpoint over cellular network interface
-                    let ping;
+                    // curl test endpoint over cellular network interface
+                    let curl;
 
                     await this.utils.waitUntil(async () => {
-                        ping = await this.worker.executeCommandInHostOS(
-                            `ping -4 -c 10 -I ${iface} ${config.network.testUrl}`,
+                        curl = await this.worker.executeCommandInHostOS(
+                            `curl -I -sS -o /dev/null -w "%{http_code}" --interface ${iface} ${URL_TEST}`,
```

Comment:
Since we are in a waitUntil loop, should we hardcode some kind of timeout? I assume the default curl timeout is much longer than a ping timeout, so we should expose it clearly in the command and maybe shorten it a bunch.

Reply:
Good thought, I would say `--connect-timeout', '15'` and `'--keepalive-time', '15',`
3 retries each? 15 second intervals?

Reply:
The default ping timeout is 1000 milliseconds (1 second) in most operating systems, so let's try 5 with curl?

Do we want/need the curl retries when it's already wrapped in a waitUntil loop? If we do both, we should make it clear via comments how many retries are expected in total.

Reply:
The `3 retries` I mentioned are for the waitUntil loop configured with `PING_ATTEMPTS`, not adding more with the curl command itself. A 5 second timeout and keepalive sounds okay to me for a response from the configured URL